mchf-eclipse/drivers/audio/audio_nr.h

```diff
@@ -28,26 +28,35 @@ typedef struct __MCHF_SPECIALMEM NoiseReduction
 	float32_t 					last_sample_buffer_L [NR_FFT_L / 2];
 	float32_t 					Hk[NR_FFT_L / 2]; // gain factors
 	float32_t 					FFT_buffer[NR_FFT_L * 2];
-	float32_t 					X[NR_FFT_L / 2][2]; // magnitudes of the current and the last FFT bins
 	float32_t 					Nest[NR_FFT_L / 2][2]; // noise estimates for the current and the last FFT frame
 	float32_t 					vk; // saved 0.24kbytes
 	float32_t 					SNR_prio[NR_FFT_L / 2];
 	float32_t 					SNR_post[NR_FFT_L / 2];
 	float32_t 					SNR_post_pos; // saved 0.24kbytes
 	float32_t 					Hk_old[NR_FFT_L / 2];
 	float32_t 					VAD;
+	int16_t						gain_display; // 0 = do not display gains, 1 = display bin gain in spectrum display, 2 = display long_tone_gain
+	//											 3 = display bin gain multiplied with long_tone_gain
+} NoiseReduction;
+
+NoiseReduction NR;
+
```

Comment:
A variable does have a memory location and here you can instruct the linker to put it into the __MCHF_SPECIALMEM.
Then it not only compiles but also works as expected. You may get an error from Travis if you have put a little too much into this struct, then carefully move some data over so that you get below the limit.

Reply:
Perfect! Thanks a lot! Finally I get it into my head :-).
Changed the code and will make another pull request.